testbed/testbed/mock_backend.go

```diff
@@ -148,6 +158,67 @@ func (mb *MockBackend) ConsumeMetric(md pdata.Metrics) {
 	if mb.isRecording {
 		mb.ReceivedMetrics = append(mb.ReceivedMetrics, md)
 	}
+	if mb.isScraping {
+		mb.getTimestamps(md)
+	}
+}
+
+func (mb *MockBackend) getTimestamps(md pdata.Metrics) {
```

Comment:
I'm not terribly comfortable with this method having side effects that aren't apparent from its name.  Perhaps this could simply discover and return the `currentTimestamp` value and the last few lines could be performed instead in the `ConsumeMetric()` method where this one is invoked.

Reply:
Thanks, I have made this change